packages/tasks/src/hardware.ts

```diff
@@ -0,0 +1,309 @@
+export const TFLOPS_THRESHOLD_WHITE_HOUSE_MODEL_TRAINING_TOTAL_BIOLOGY = 10 ** 11;
+export const TFLOPS_THRESHOLD_WHITE_HOUSE_CLUSTER = 10 ** 8;
+
+/**
+ * EU AI Act
+ * https://ec.europa.eu/commission/presscorner/detail/en/qanda_21_1683
+ */
+export const TFLOPS_THRESHOLD_EU_AI_ACT_MODEL_TRAINING_TOTAL = 10 ** 13;
+
+export interface HardwareSpec {
+	/**
+	 * Approximate value, in FP16 whenever possible.
+	 * This is only approximate/theoretical and shouldn't be taken too seriously.
+	 * Currently the CPU values are from cpu-monkey.com
+	 * while the GPU values are from techpowerup.com
+	 *
+	 * Note to reviewers: I got fed up with data entry,
+	 * and HuggingChat running Llama3 with Web search was failing a bit,
+	 * so some of those values might be slightly inaccurate. Forgive me and please feel free to improve.
+	 */
+	tflops: number;
+	/**
+	 * If an array is specified, options of memory size (can be VRAM, unified RAM)
+	 * e.g. an A100 exists in 40 or 80 GB.
+	 */
+	memory?: number[];
+}
+
+export const DEFAULT_MEMORY_OPTIONS = [8, 16, 24, 32, 40, 48, 64, 80, 96, 128];
+
+export const SKUS = {
+	GPU: {
+		NVIDIA: {
+			H100: {
+				tflops: 267.6,
+				memory: [80],
+			},
+			L40: {
+				tflops: 90.52,
+				memory: [48],
+			},
+			"RTX 6000 Ada": {
+				tflops: 91.1,
+				memory: [48],
+			},
+			A100: {
+				tflops: 77.97,
+				memory: [80, 40],
+			},
+			A40: {
+				tflops: 37.42,
+				memory: [48],
+			},
+			A10: {
+				tflops: 31.24,
+				memory: [24],
+			},
+			T4: {
+				tflops: 65.13,
+				memory: [16],
+			},
+			"RTX 4090": {
+				tflops: 82.58,
+				memory: [24],
+			},
+			"RTX 4080 SUPER": {
+				tflops: 52.2,
+				memory: [16],
+			},
+			"RTX 4080": {
+				tflops: 48.7,
+				memory: [16],
+			},
+			"RTX 4070": {
+				tflops: 29.15,
+				memory: [12],
+			},
+			"RTX 4070 Ti": {
+				tflops: 40.09,
+				memory: [12],
+			},
+			"RTX 4070 Super": {
+				tflops: 35.48,
+				memory: [12],
+			},
+			"RTX 4070 Ti Super": {
+				tflops: 44.1,
+				memory: [16],
+			},
+			"RTX 3090": {
+				tflops: 35.58,
+				memory: [24],
+			},
+			"RTX 3090 Ti": {
+				tflops: 40,
+				memory: [24],
+			},
+			"RTX 3080 Ti": {
+				tflops: 34.1,
+				memory: [12],
+			},
+			"RTX 3080": {
+				tflops: 30.6,
+				memory: [12, 10],
+			},
+		},
+		AMD: {
+			MI300: {
+				tflops: 383.0,
+				memory: [192],
+			},
+			MI250: {
+				tflops: 362.1,
+				memory: [128],
+			},
+			MI210: {
+				tflops: 181.0,
+				memory: [64],
+			},
+			"RX 7900 XTX": {
+				tflops: 122.8,
+				memory: [24],
+			},
+			"RX 7900 XT": {
+				tflops: 103.0,
+				memory: [20],
+			},
+			"RX 7900 GRE": {
+				tflops: 91.96,
+				memory: [16],
+			},
+			"RX 7800 XT": {
+				tflops: 74.65,
+				memory: [16],
+			},
+			"RX 7700 XT": {
+				tflops: 70.34,
+				memory: [12],
+			},
+			"RX 7600 XT": {
+				tflops: 45.14,
+				memory: [16, 8],
+			},
+		},
+	},
+	CPU: {
+		Intel: {
+			"Xeon 4th Generation (Sapphire Rapids)": {
+				tflops: 3.8,
```

Comment:
```suggestion
				tflops: 1.3,
```
Sauce: https://www.intel.com/content/dam/support/us/en/documents/processors/APP-for-Intel-Xeon-Processors.pdf